File: src/main/java/org/alham/alhamfirst/mapper/TodoMapper.java
package org.alham.alhamfirst.mapper;

import org.alham.alhamfirst.dto.todo.TodoDTO;
import org.alham.alhamfirst.entity.todo.Todo;
import org.springframework.stereotype.Component;

@Component
public class TodoMapper {

    public Todo createTodoFromDTO(TodoDTO todoDTO) {
        return Todo.builder()
                .detail(todoDTO.getDetail())
                .completed(todoDTO.isCompleted())
                .startDate(todoDTO.getStartDate())
                .build();
    }

    public TodoDTO createTodoDTOFromEntity(Todo todo) {
        return TodoDTO.builder()
                .id(todo.getId())
                .detail(todo.getDetail())
                .completed(todo.isCompleted())
                .build();
    }




    public void updateTodoFromDTO(Todo todo, TodoDTO todoDTO) {
        todo.updateTodo(todoDTO.getDetail());
    }
}

File: src/main/java/org/alham/alhamfirst/mapper/TodoMapper.kt
package org.alham.alhamfirst.mapper;

import org.alham.alhamfirst.domain.dto.todo.TodoDTO
import org.alham.alhamfirst.domain.entity.todo.Todo

class TodoMapper {


    fun createTodoFromDTO(todoDTO: TodoDTO) : Todo {
        return Todo(
            detail=todoDTO.detail,
            completed = todoDTO.completed,
            startDate = todoDTO.startDate
        );
    }

    fun createTodoDTOFromEntity(todo : Todo): TodoDTO {
        return TodoDTO(
            id = todo.id,
            detail = todo.detail,
            completed = todo.completed,
        )
    }

    /**
     * userId를 포함한 TodoDTO 생성
     * @param todo
     * @return
     */

    fun createTodoDTOWithUserIdFromEntity(todo : Todo): TodoDTO {
        return TodoDTO(
            id = todo.id,
            userId = todo.user?.id,
            detail = todo.detail,
            completed = todo.completed
        )
    }


    fun updateTodoFromDTO(todo : Todo, todoDTO : TodoDTO) {
        todo.updateTodo(
            todoDTO.detail
        )
    }

}
